File: CppPractice/CppPractice/unique_morse_code_words.cpp
#pragma once
#include "unique_morse_code_words.h"

int Solution::uniqueMorseRepresentations(vector<string>& words) {
	const string morseMap[] = { ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.." };
	map<string, bool> resultMap;
	for (int i = 0; i < words.size(); i++) {
		string line = "";
		for (int j = 0; j < words[i].size(); j++) {
			line = line + morseMap[words[i][j] - 'a'];
		}
		resultMap[line] = true;
	}
	return resultMap.size();
}

File: JavaPractice/src/leetcode/UniqueMorseCodeWords.java
package leetcode;

import java.util.HashMap;

public class UniqueMorseCodeWords {
    class Solution {
        private final String[] MORSE_MAP = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        public int uniqueMorseRepresentations(String[] words) {
            HashMap<String,Boolean> resultMap = new HashMap<>();
            for(String line:words){
                StringBuilder morseBuilder = new StringBuilder();
                for(int i=0; i < line.length(); i++){
                    char tmpChar = line.charAt(i);
                    morseBuilder.append(MORSE_MAP[tmpChar-'a']);
                }
                resultMap.put(morseBuilder.toString(), true);
            }
            return resultMap.size();
        }
    }
}
